### PR TITLE
Ref als help

### PR DESCRIPTION
**Pull request by kenbowen**
The ref_als_help branch now has the newly revised version of the alshelp system.  It runs by opening als_help.html in the same directory as the folder alshelp/ subfolder, and the files alshelp.css and package_nav.html.  There are 4 packages: Core Prolog, ALSDev, ALS Library, and  ALS C-Interf, but only the 1st two have any content.  [The reason for adding the SAVE_PRE_REV.zip is that the conversation machinery may be useful in developing machinery to populate ALS Library and ALS C-Interf.]   There's way too many pages to actually check thru all now, but please give it an overall look-over to see if there are any glaring faults.

### CODE REVIEW

File: manual/als_help.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<HTML>
+<HEAD>
+<TITLE>ALS Prolog Help</TITLE>
+<link rel="stylesheet" type="text/css" href="alshelp.css">
+<HEAD>
+<style>
```
Comment (chuckhoupt) [inline manual/als_help.html]: Might as well remove this inline style, since they are unused.

File: manual/als_help.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#menu li {
+    list-style-type:none;
+}
+</style>
+</HEAD>
+<BODY CLASS="body">
```
Comment (chuckhoupt) [inline manual/als_help.html]: There are no styles assigned to the class "body", so might as well remove.

File: manual/alshelp.css [hunk 3]
```diff
@@ -0,0 +1,68 @@
+#body {
```
Comment (chuckhoupt) [inline manual/alshelp.css]: There is no element of id "body", so might as well remove.

File: manual/als_help.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  <IFRAME SRC="package_nav.html"  id="left_top" CLASS="left_top" title="alspackages" scrolling="no"></IFRAME>
```
Comment (chuckhoupt) [inline manual/als_help.html]: Matching IDs and CLASSes is confusing. Only the IDs are referenced by styles or used by TOC, so remove CLASSes.

The SCROLLING attribute is deprecated, but can be replaced with CSS.

You might consider using semantic ID names rather then positional names (e.g. packages, toc, content).

File: manual/alshelp.css [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+position: absolute;
+top:0;
+left:0;
+height:8em;
+width:13em;
+background-color:ivory;
+}
+
+#left_bottom {
+position: absolute;;
```
Comment (chuckhoupt) [inline manual/alshelp.css]: Extra semi-colon.

File: manual/alshelp.css [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+top:8em;
+left:0;
+height:100%;
+width:13em;
+background-color:cornsilk;
+}
+
+#right {
+position: absolute;;
```
Comment (chuckhoupt) [inline manual/alshelp.css]: extra semi-colon.

File: manual/alshelp.css [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+top:0;
+left:13em;
+height:120%;
```
Comment (chuckhoupt) [inline manual/alshelp.css]: The 120% height and width setting causes problem as you resize the window. (sometime the "right" frames scroll bar is visible, sometime hidden, etc).

I'd recommend removing the height/width and then setting top, right, bottom to 0 (keeping left at 13em).

File: manual/alshelp.css [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+width:80%;
+background-color:ivory;
+}
+
+ul {
```
Comment (chuckhoupt) [inline manual/alshelp.css]: All the styles from here on are unused and could be removed.

File: manual/alshelp.css (hunk 7 above)
Comment (kenbowen) [inline manual/alshelp.css]: I just forgot to clean out most of alshelp.css � some junk, some moved into the alshelp/toc_ABC.html files.

Re: I'd recommend removing the height/width and then setting top, right, bottom to 0 (keeping left at 13em).

This doesn�t appear to work.  When I tried it, things scrunched up & to the left.

� Ken

Comment (chuckhoupt) [inline manual/alshelp.css]: Looks like iframes don't interact well with position fixed/absolute. The work-around is to wrap them in divs:

http://stackoverflow.com/questions/4195971/iframe-with-a-css-fixed-position-possible

**Comment by chuckhoupt** [comment]
I committed a fix for the iframe layout problems: e3a4cf339ed82fa03c3b7b5d636933c89dbf415e

**Comment by kenbowen** [comment]
Thanks!

**Pull request merged**